WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+import { Button } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+import '../../stylesheets/delegates/style.scss';
+
+export default function Delegates({
+  delegates,
+  isEditVisible,
+}) {
+  const [seniorDelegates, setSeniorDelegates] = React.useState([]);
+  const [delegatesUnderSeniorDelegates, setDelegatesUnderSeniorDelegates] = React.useState([]);
+
+  // TO_VERIFY: I assume there are no cases where there are no delegates without
+  // senior delegate unless they are senior delegate
+
+  React.useEffect(() => {
+    setSeniorDelegates(delegates
+      .filter((user) => user.delegate_status === 'senior_delegate')
+      .sort((user1, user2) => {
+        if (user1.region < user2.region) {
+          return -1;
+        } if (user1.region > user2.region) {
+          return 1;
+        }
+        return 0;
+      }));
+  }, []);
+
+  React.useEffect(() => {
+    setDelegatesUnderSeniorDelegates(seniorDelegates
+      .map((seniorDelegate) => [seniorDelegate, ...delegates
+        .filter((user) => user.senior_delegate_id === seniorDelegate.id && user.delegate_status !== 'trainee_delegate')
+        .sort((user1, user2) => {
+          if (user1.region < user2.region) {
+            return -1;
+          } if (user1.region > user2.region) {
+            return 1;
+          } if (user1.name < user2.name) {
+            return -1;
+          } if (user1.name > user2.name) {
+            return 1;
+          }
+          return 0;
+        })]));
+  }, [seniorDelegates]);
+
+  const dasherize = (string) => string.replace(/_/g, '-');
+
+  return seniorDelegates.map((seniorDelegate, index) => (
+    <div
+      className="table-responsive"
+      key={`region-${seniorDelegate.id}`}
+    >
+      <table className="table delegates-table">
+        <colgroup>
+          <col />
+          <col className="col-md-4" />
+          <col className="col-md-4" />
+          <col className="col-md-4" />
+        </colgroup>
+        <thead>
+          <tr>
+            <th />
+            <th>{I18n.t('delegates_page.table.name')}</th>
+            <th>{I18n.t('delegates_page.table.role')}</th>
+            <th>{I18n.t('delegates_page.table.region')}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {!!delegatesUnderSeniorDelegates[index]
```

Comment:
Could this ever be false?

Reply:
Yes in initial load, but with above changes, it is no more required. :-)